scripts/process_modis.py

```diff
@@ -0,0 +1,399 @@
+    ds["TIWP_global_distribution_unmasked"] /= TIWP_global_distribution_unmasked_count
+    ds["CF_global_distribution_masked"] /= CF_global_distribution_masked_count
+    ds["TIWP_global_distribution_masked"] /= TIWP_global_distribution_masked_count
+
+    return ds
+
+
+def process_zonal_means(directory_path, mask):
+    """
+    Calculate zonal means of TIWP and CF over all years of modis data.
+    """
+
+    suffix_for_variable = "masked" if mask else "unmasked"
+
+    lat = np.arange(-89.5, 90.5, 1)
+    n_lats = 180
+
+    CF_zonal_mean_unmasked_count = np.zeros((n_lats))
+    TIWP_zonal_mean_unmasked_count = np.zeros((n_lats))
+    CF_zonal_mean_masked_count = np.zeros((n_lats))
+    TIWP_zonal_mean_masked_count = np.zeros((n_lats))
+
+    ds = xr.Dataset(
+        {
+            "CF_zonal_mean_unmasked": (
+                ("lat"),
+                np.zeros((n_lats)),
+            ),
+            "TIWP_zonal_mean_unmasked": (
+                ("lat"),
+                np.zeros((n_lats)),
+            ),
+            "CF_zonal_mean_masked": (
+                ("lat"),
+                np.zeros((n_lats)),
+            ),
+            "TIWP_zonal_mean_masked": (
+                ("lat"),
+                np.zeros((n_lats)),
+            ),
+        },
+        coords={
+            "lat": lat,
+        },
+    )
+
+    mask = mask.mask == 1
+
+    for file in os.listdir(directory_path):
+        if file.endswith(".nc"):
+            file_path = os.path.join(directory_path, file)
+            print(file_path)
+            if "_masked.nc" in file_path:
+                continue
+            if "global" in file_path:
+                continue
+            if "time" in file_path:
+                continue
+            if "zonal" in file_path:
+                continue
+            for masked in [True, False]:
+                suffix_for_variable = "masked" if masked else "unmasked"
+                dataset = xr.open_dataset(file_path)
+                if masked:
+                    for var in dataset.data_vars:
+                        dataset[var] = dataset[var].where(mask == 1)
+
+                # remove failed downloads (negative date)
+                dataset = dataset.where(dataset["date"] >= 0, drop=True)
+
+                IWP = dataset.Cloud_Water_Path_Ice_Mean
+                IWP = IWP.where(IWP >= 0)
+
+                IWP_counts = dataset.Cloud_Retrieval_Fraction_Ice_Pixel_Counts
+                IWP_counts = IWP_counts.where(IWP_counts >= 0)
+
+                CF_ice = dataset.Cloud_Retrieval_Fraction_Ice * 0.0001
+                CF_ice = CF_ice.where(CF_ice >= 0)
+
+                CF = dataset.Cloud_Fraction_Mean * 0.0001
+                CF = CF.where(CF >= 0)
+                CF_counts = dataset.Cloud_Fraction_Pixel_Counts
+                CF_counts = CF_counts.where(CF_counts >= 0)
+
+                IWP = IWP.where(IWP >= 0).fillna(0)
+                TIWP = IWP * CF_ice
```

Comment:
I'm guessing because the unit is g/m^{2}, and they don't retrieve less than 1 g/m^{2}.
"_Cloud_Water_Path_Ice_Mean:units = "g/m^2" ;
Cloud_Water_Path_Ice_Mean:valid_range = 0s, 6000s ;
Cloud_Water_Path_Ice_Mean:_FillValue = -9999s ;
Cloud_Water_Path_Ice_Mean:scale_factor = 1.0d ;_"

It also comes as an int in the original data. (Actually an int16, now that I check again, so I could have saved some storage space here).